doc/lag/Increasing LACP PDU timeout during warm-reboot.md

```diff
@@ -0,0 +1,119 @@
+below doesn't wait for an acknowledgment packet, there's no reliable way of
+knowing whether it was handled by the peer device or not.
+
+# Background
+
+LACP supports two rates for sending PDUs. There is a short rate, where a PDU is
+sent every 1 second, and a long rate, where a PDU is sent every 30 seconds. Both
+sides know what rate to expect from the other side. If 3 LACP PDUs are missed,
+then the LAG is considered to be down, and data traffic is stopped. This results
+in an effective timeout of 3 seconds when using the short rate and 90 seconds
+when using the long rate.
+
+# Protocol
+
+To change the number of retries, an Ethernet packet of the fillowing structure
+will be sent. This Ethernet packet will have an ethertype of 0x6300, and will
+not have an IPv4 or IPv6 layer on top of it. Instead, there will instead be
+multiple TLV fields, similar to LACP.
+
+The TLV types will be defined as follows:
+
+| Value | Description         |
+|-------|---------------------|
+| 0x01  | Actor Information   |
+| 0x02  | Partner Information |
+| 0x03  | Retry Count         |
+
+Both Actor Information and Partner Information have the following content:
+
+| Starting byte | Length | Description     |
+|---------------|--------|-----------------|
+|      0        |   2    | System Priority |
+|      2        |   6    | System ID       |
+|      8        |   2    | Key             |
+|     10        |   2    | Port Priority   |
+|     12        |   2    | Port            |
+|     14        |   2    | Padding         |
+
+Retry count have the following content:
+
+| Starting byte | Length | Description     |
+|---------------|--------|-----------------|
+|      0        |   2    | New retry count |
+|      2        |   2    | Padding         |
+
+When the retry count needs to be changed, the sending device must send a packet
+with ethertype 0x6300, and the data will contain the Actor Information, Partner
+Information, and Retry Count TLVs. The receiving device must validate the actor
+and partner information, and then update the retry count as specified. No
+acknowledgment packet is sent back.
+
+The custom ethertype is so that if the packet is sent to non-SONiC devices
+or SONiC devices running an older version, then it can be silently discarded,
+without it getting incorrectly handled by the peer device and without it getting
+forwarded to a different device.
+
+# Changing Max Retries for Warmboot
+
+As part of a SONiC device starting the warmboot process, LACP PDUs are sent to
+all of the peers, to refresh the timers on the peers. This allows the warmboot
+process the full 90 seconds for control plane to come back up and for PDUs to be
+sent again after warmboot.
+
+Now, in addition to refreshing the PDUs timer, the above-specified Ethernet
+packet (with ethertype 0x6300) will be sent to the peer devices, with the new
+retry count set to 5. This notifies the peer device that for this device that
```

Comment:
Why cant we make this retry-count as user configurable with default value 5?

Reply:
The HLD has been updated with a new design/approach. The retry count is now user-configurable.